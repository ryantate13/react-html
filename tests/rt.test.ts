import {rt} from '../src';

describe('rt', () => {
    it('renders with children', () => {
        expect(rt({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(rt({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(rt()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(rt({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(rt({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(rt({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(rt({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(rt({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(rt({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(rt({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(rt({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(rt({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(rt({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(rt({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(rt({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(rt({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(rt({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(rt({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(rt({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(rt({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(rt({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(rt({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(rt({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(rt({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(rt({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(rt({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(rt({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(rt({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(rt({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(rt({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(rt({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(rt({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(rt({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(rt({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(rt({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(rt({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(rt({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(rt({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(rt({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(rt({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(rt({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(rt({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(rt({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(rt({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(rt({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(rt({translate: "no"})).toMatchSnapshot();
    });
});