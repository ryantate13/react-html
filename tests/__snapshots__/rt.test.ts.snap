// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rt renders arrays of children 1`] = `
<rt>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</rt>
`;

exports[`rt renders with accessKey="a,b,c,d" 1`] = `
<rt
  accessKey="a,b,c,d"
/>
`;

exports[`rt renders with autoCapitalize="characters" 1`] = `
<rt
  autoCapitalize="characters"
/>
`;

exports[`rt renders with autoCapitalize="none" 1`] = `
<rt
  autoCapitalize="none"
/>
`;

exports[`rt renders with autoCapitalize="off" 1`] = `
<rt
  autoCapitalize="off"
/>
`;

exports[`rt renders with autoCapitalize="on" 1`] = `
<rt
  autoCapitalize="on"
/>
`;

exports[`rt renders with autoCapitalize="sentences" 1`] = `
<rt
  autoCapitalize="sentences"
/>
`;

exports[`rt renders with autoCapitalize="words" 1`] = `
<rt
  autoCapitalize="words"
/>
`;

exports[`rt renders with children 1`] = `
<rt>
  TEST
</rt>
`;

exports[`rt renders with className="foo bar baz" 1`] = `
<rt
  className="foo bar baz"
/>
`;

exports[`rt renders with contentEditable="false" 1`] = `
<rt
  contentEditable="false"
/>
`;

exports[`rt renders with contentEditable="true" 1`] = `
<rt
  contentEditable="true"
/>
`;

exports[`rt renders with dir="auto" 1`] = `
<rt
  dir="auto"
/>
`;

exports[`rt renders with dir="ltr" 1`] = `
<rt
  dir="ltr"
/>
`;

exports[`rt renders with dir="rtl" 1`] = `
<rt
  dir="rtl"
/>
`;

exports[`rt renders with draggable="false" 1`] = `
<rt
  draggable="false"
/>
`;

exports[`rt renders with draggable="true" 1`] = `
<rt
  draggable="true"
/>
`;

exports[`rt renders with empty attributes 1`] = `<rt />`;

exports[`rt renders with hidden=false 1`] = `
<rt
  hidden={false}
/>
`;

exports[`rt renders with hidden=true 1`] = `
<rt
  hidden={true}
/>
`;

exports[`rt renders with id="TEST" 1`] = `
<rt
  id="TEST"
/>
`;

exports[`rt renders with inputMode="decimal" 1`] = `
<rt
  inputMode="decimal"
/>
`;

exports[`rt renders with inputMode="email" 1`] = `
<rt
  inputMode="email"
/>
`;

exports[`rt renders with inputMode="none" 1`] = `
<rt
  inputMode="none"
/>
`;

exports[`rt renders with inputMode="numeric" 1`] = `
<rt
  inputMode="numeric"
/>
`;

exports[`rt renders with inputMode="search" 1`] = `
<rt
  inputMode="search"
/>
`;

exports[`rt renders with inputMode="tel" 1`] = `
<rt
  inputMode="tel"
/>
`;

exports[`rt renders with inputMode="text" 1`] = `
<rt
  inputMode="text"
/>
`;

exports[`rt renders with inputMode="url" 1`] = `
<rt
  inputMode="url"
/>
`;

exports[`rt renders with is="foo" 1`] = `
<rt
  is="foo"
/>
`;

exports[`rt renders with itemID="https://google.com/" 1`] = `
<rt
  itemID="https://google.com/"
/>
`;

exports[`rt renders with itemProp="https://google.com/" 1`] = `
<rt
  itemProp="https://google.com/"
/>
`;

exports[`rt renders with itemRef="foo bar" 1`] = `
<rt
  itemRef="foo bar"
/>
`;

exports[`rt renders with itemScope=false 1`] = `
<rt
  itemScope={false}
/>
`;

exports[`rt renders with itemScope=true 1`] = `
<rt
  itemScope={true}
/>
`;

exports[`rt renders with itemType="https://google.com/" 1`] = `
<rt
  itemType="https://google.com/"
/>
`;

exports[`rt renders with lang="" 1`] = `
<rt
  lang=""
/>
`;

exports[`rt renders with lang="es-419" 1`] = `
<rt
  lang="es-419"
/>
`;

exports[`rt renders with slot="TEST" 1`] = `
<rt
  slot="TEST"
/>
`;

exports[`rt renders with spellCheck="false" 1`] = `
<rt
  spellCheck="false"
/>
`;

exports[`rt renders with spellCheck="true" 1`] = `
<rt
  spellCheck="true"
/>
`;

exports[`rt renders with style={"color":"#000000"} 1`] = `
<rt
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`rt renders with tabIndex=1 1`] = `
<rt
  tabIndex={1}
/>
`;

exports[`rt renders with title="TEST" 1`] = `
<rt
  title="TEST"
/>
`;

exports[`rt renders with translate="no" 1`] = `
<rt
  translate="no"
/>
`;

exports[`rt renders with translate="yes" 1`] = `
<rt
  translate="yes"
/>
`;

exports[`rt renders without attributes 1`] = `<rt />`;
